proxy/context.go

```diff
@@ -206,6 +210,38 @@ func (c *context) Metrics() filters.Metrics            { return c.metrics }
 func (c *context) Tracer() opentracing.Tracer          { return c.tracer }
 func (c *context) ParentSpan() opentracing.Span        { return c.parentSpan }
 
+func (c *context) Logger() *log.Entry {
+	traceId := c.getTraceID()
+	return log.WithFields(log.Fields{"trace_id": traceId})
+}
+
+func (c *context) getTraceID() string {
```

Comment:
Lets do the following:
- return empty value from `getTraceID` if we can not obtain trace id
- return log.WithFields only if traceId is not empty, otherwise fallback to the default log